.github/workflows/proselint.yml

```diff
@@ -0,0 +1,56 @@
+# Copyright (C) 2022 Daniel Stenberg, <daniel@haxx.se>, et al.
+#
+# SPDX-License-Identifier: curl
+
+name: proselint
+
+on:
+  push:
+    branches:
+    - master
+    - '*/ci'
+  pull_request:
+    branches:
+    - master
+
+jobs:
+  check:
+
+    runs-on: ubuntu-latest
+
+    steps:
+    - uses: actions/checkout@v2
+
+    - name: install prereqs
+      run: sudo apt-get install python3-proselint
+
+    # config file help: https://github.com/amperser/proselint/
+    - name: create proselint config
+      run: grep '^## C:' .github/workflows/proselint.yml | sed 's/^## C://' > $HOME/.proselintrc
+
+    - name: check prose
+      run: a=`git ls-files '*.md' | grep -v docs/CHECKSRC.md` && proselint $a README
+
+    - name: create second proselint config
+      run: grep '^## C2:' .github/workflows/proselint.yml | sed 's/^## C2://' > $HOME/.proselintrc
+
+    - name: check special prose
+      run: a=docs/CHECKSRC.md && proselint $a
+
+## C: {
```

Comment:
This just feels like a kludge. I'd put this in something like .github/proselint/special.json, then have a 
```
cp .github/proselint/special.json ~/.config/proselint/config.json
```
(this is the new path for proselint as per https://github.com/amperser/proselint#checks )

Reply:
I opted to not create a stand-alone json file ingit because they don't have comments and then I can't easily provide copyright info and would need to have yet another exception for that in `.reuse/dep5` and I rather not.

Reply:
Ok - could something like this work instead?

```
max@LAPTOP-BL9ORR19:~$ cat special.yml
# I have comments because I am a YAML file
{
  "checks": {
    "typography.diacritical_marks": false,
    "typography.symbols": false,
    "annotations.misc": false,
  }
}
max@LAPTOP-BL9ORR19:~$ grep -v "#" special.yml
{
  "checks": {
    "typography.diacritical_marks": false,
    "typography.symbols": false,
    "annotations.misc": false,
  }
}
max@LAPTOP-BL9ORR19:~$ yq . special.yml
{
  "checks": {
    "typography.diacritical_marks": false,
    "typography.symbols": false,
    "annotations.misc": false
  }
}
```

YAML is a superset of JSON - you can write the necessary copyright comments with lines starting #, then either grep them out or use `yq` (would need installing along with `jq`) to parse and output the JSON content.

Reply:
But if we have to run grep anyway, does it really matter?

Reply:
Since we have to copy the file to a different place anyway, we can just call i ".json" and still do the grep and allow `#`-comments...

Reply:
That doesn't really solve that we need two different though, so we'd need to do two different files I guess. Or do something crazier.

Reply:
Ideally the benefits here comes from having editor tooling which understands how to edit JSON and YAML. If more options need adding in future then having a JSON/YAML aware editor means that people will make fewer mistakes; it's also more obvious to people where changes need to be added and how to add them. 

IMO we should be generally aiming (in all aspects) for "least surprise" to developers - if I were a new dev coming into this project and wanted to add a setting to proselint for some reason, I'd expect that to be mastered in its own file as opposed to mixed in with the CI definition. 

There are reasons for wanting to put significant chunks of code in CI definitions (e.g. they're being shared between projects) but I don't think this is one of them.

Anyway, I'm not going to argue this further! It's only my recommendation.

Reply:
If I make two files called `proselint.json` and `proselint2.json` for this, it still requires some good insight for a developer to know how to update them and to what, and probably they need to read the `proselint.yml` file to learn that, and then they have the whole picture anyway...

Reply:
I'll update to that and see how that feels in comparison.

Reply:
You could also just include the config in each file creation step with a multi-line (using | in YAML) shell command to use echo or tee.